lib/queuePopulator/KafkaLogConsumer/LogConsumer.js

```diff
@@ -0,0 +1,166 @@
+            this._consumer.subscribe([this._topic]);
+            return done();
+        });
+    }
+
+    /**
+     * Waits for consumer group to rebalance
+     * @param {number} wait amount of time to wait
+     * @param {Function} cb callback
+     * @returns {undefined}
+     */
+    _waitForAssignment(wait, cb) {
+        setTimeout(() => {
+            /**
+             * assignements contain the partitions
+             * assigned to this consumer, they are only
+             * set once consumer group is balanced
+             */
+            const assignments = this._consumer.assignments();
+            if (assignments.length === 0) {
+                if (wait > maxDelayRebalance) {
+                    return cb(true);
+                }
+                return this._waitForAssignment(wait + 2000, cb);
+            }
+            return cb();
+        }, 2000);
+    }
+
+    /**
+     * Queries last commited partition offsets
+     * and stores them
+     * @param {Function} cb callback
+     * @returns {undefined}
+     */
+    _storeCurrentOffsets(cb) {
+        this._consumer.committed(5000, (err, topicPartitions) => {
+            if (err) {
+                this._log.error('Error while getting offsets', {
+                    method: 'LogConsumer._storeCurrentOffsets',
+                    topic: this._topic,
+                    groupId: this._groupId,
+                });
+                return cb(err);
+            }
+            // saving offsets
+            this._offsets = topicPartitions;
+            return cb();
+        });
+    }
+
+    /**
+     * Inintializes record stream
+     * @returns {undefined}
+     */
+    _resetRecordStream() {
+        this._listRecordStream = new ListRecordStream(this._log);
+        this._listRecordStream.getOffset = this._getOffset.bind(this);
+    }
+
+    /**
+     * Consumes kafka messages and writes them to record
+     * stream
+     * @param {Number} limit maximum messages to consume
+     * @param {Function} cb callback
+     * @returns {undefined}
+     */
+    _consumeKafkaMessages(limit, cb) {
+        this._resetRecordStream();
+        this._consumer.consume(limit, (err, messages) => {
+            if (err) {
+                this._log.error('Error occured while consuming messages', {
+                    method: 'LogConsumer.readRecords',
+                    topic: this._topic,
+                    groupId: this._groupId,
+                });
+                return cb(err);
+            }
+            // writing consumed messages to the stream
+            messages.forEach(message => this._listRecordStream.write(message));
+            return cb();
+        });
+    }
+
+    /**
+     * Reads a certain number of messages from oplog kafka topic
+     * The caller of this function expects a stream to be returned
+     * in the callback
+     * @param {Object} params reading params
+     * @param {number} params.limit maximum number of elements to fetch
```

Comment:
what is this "limit" : the size of a batch, or the total number of elements?

generally, we should not limit the (total) number of elements i guess : we just setup the stream, and process messages until stopped, no?